### PR TITLE
Add Detection for CVE-2023-28461

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**
### **Description:**

This pull request implements an exploit for **CVE-2023-28461**, a critical remote code execution vulnerability found in Array Networks Array AG Series and vxAG SSL VPN gateways. The vulnerability allows attackers to exploit an HTTP header with the `'flags'` attribute, enabling them to browse the system's filesystem without authentication.

### **Changes:**

1. **Exploit Class (`CVE202328461Exploit`)**:
    
    - The `CVE202328461Exploit` class is registered with the exploits registry and extends `webexploit.WebExploit`.
    - **Metadata**: Title, description, and reference to CVE-2023-28461 with a **CRITICAL** risk rating.
2. **SSL Context Creation**:
    
    - The SSL handling was specifically added to account for the target's potentially insecure SSL configuration, allowing the exploit to function even if the server's SSL/TLS certificates are invalid or improperly configured.
3. **SSL Request Handling**:
    
    - The `_send_ssl_request` method sends GET requests using `urllib3` with the custom SSL context. This ensures that the exploit can securely bypass any SSL/TLS certificate verification failures, which may be part of the target's security misconfiguration.
4. **Target Validation**:
    
    - The `accept` method checks if the target endpoint matches the expected pattern based on a regular expression (`Rel_AG_(\d+_\d+_\d+_\d+)`).
5. **Vulnerability Detection**:
    
    - The `check` method scans the target's response for the vulnerable version by matching the pattern and comparing the version string with `MAX_VULNERABLE_VERSION`.
    - If a vulnerable version is detected, a new vulnerability instance is created and added to the list.

image

### CODE REVIEW

**Review by ybadaoui-ostorlab — COMMENTED** [review]
can you please add a description with a screenshot poc of a real target!

File: agent/exploits/cve_2023_28461.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        ssl_context = self._create_ssl_context()
+        with urllib3.PoolManager(ssl_context=ssl_context) as http:
+            try:
+                response = http.request(
+                    "GET", url, timeout=DEFAULT_TIMEOUT.total_seconds()
+                )
+                return response.data.decode("utf-8")
+            except urllib3.exceptions.HTTPError as e:
+                logging.error("Error during SSL request: %s", e)
+                return ""
```
Comment (mohsinenar) [inline agent/exploits/cve_2023_28461.py:63]: Both methods are not using self and are not related to the class take them outside.

**Review by mohsinenar — CHANGES_REQUESTED** [review]

File: agent/exploits/cve_2023_28461.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        for pattern in self.accept_pattern:
+            match = pattern.search(response_body)
+            if match is not None:
+                version_str = match.group(1).replace("_", ".")
+                target_version = version.parse(version_str)
+
+                if target_version <= MAX_VULNERABLE_VERSION:
+                    vulnerability = self._create_vulnerability(target)
+                    vulnerabilities.append(vulnerability)
+                break
```
Comment (ohachimOs) [inline agent/exploits/cve_2023_28461.py:105]:
```suggestion
            if match is None:
               break
             version_str = match.group(1).replace("_", ".")
             target_version = version.parse(version_str)

             if target_version <= MAX_VULNERABLE_VERSION:
                 vulnerability = self._create_vulnerability(target)
                 vulnerabilities.append(vulnerability)
               ```

Comment (nmasdoufi-ol) [reply agent/exploits/cve_2023_28461.py:105]: OK

File: tests/exploits/cve_2023_28461_test.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    requests_mock.get(
+        "http://localhost:80/prx/000/http/localhost/login",
+        text=NO_VERSION_RESPONSE,
+        status_code=200,
+    )
+
+    with patch(
+        "agent.exploits.cve_2023_28461._send_ssl_request",
+        return_value=NO_VERSION_RESPONSE,
+    ):
```
Comment (adnaneserrar) [inline tests/exploits/cve_2023_28461_test.py:105]: agent_exploit=mocker.patch("agent.exploits.cve_2023_28461._send_ssl_request",return_value=NO_VERSION_RESPONSE)

**Review by oussamaessaji — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**